pallets/liquid-staking-two/src/lib.rs

```diff
@@ -247,45 +249,116 @@ pub mod pallet {
             origin: OriginFor<T>,
             era_index: EraIndex,
         ) -> DispatchResultWithPostInfo {
+            ensure_signed(origin)?; //TODO(Alan): Who can call this extrinsic?
+            CurrentEra::<T>::put(era_index);
+            // TODO(Alan): Should we deposit event?
             Ok(().into())
         }
 
-        //todo，record reward on each era, invoked by stake-client
+        /// Invoked by offchain stake client. Declare reward received.
+        ///
+        /// `origin` should be approved account.(not implement yet).
+        ///
+        /// Q: What's the agent?
+        /// Q: Should update exchange_rate?
         #[pallet::weight(10_000)]
         #[transactional]
-        pub fn record_reward(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
-            Ok(().into())
-        }
-
-        //todo invoked by stake-client, considering insurrance pool
-        #[pallet::weight(10_000)]
-        #[transactional]
-        pub fn record_slash(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
-            Ok(().into())
-        }
-
-        // bond/unbond/rebond/bond_extra may be merge into one
-        #[pallet::weight(10_000)]
-        #[transactional]
-        pub fn record_bond_response(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
-            Ok(().into())
-        }
-
-        #[pallet::weight(10_000)]
-        #[transactional]
-        pub fn record_bond_extra_response(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+        pub fn record_reward(
+            origin: OriginFor<T>,
+            agent: T::AccountId,
+            #[pallet::compact] amount: Balance,
+            block_number: BlockNumber,
+        ) -> DispatchResultWithPostInfo {
+            let _sender = ensure_signed(origin)?;
+            MatchingPool::<T>::try_mutate(
```

Comment:
`MatchingPool` mainly works when user stake and unstake, please refer to :
https://github.com/parallel-finance/parallel/blob/master/pallets/liquid-staking/src/lib.rs#L365

when record_reward or record_slash, exchangerate will be updated.

excahngerate = stakingcurrency amount/ liquidcurrency amount；
the amount of stakingcurrency(KSM) will be store in `StakingPool`,
the amount of liquidcurrency(xKSM) can get like this: T::Currency::free_balance()

Reply:
free_balance of whom? The agent?

Reply:
should be T::Currency::total_issuance()